src/lava/utils/dataloader/mnist.py

```diff
@@ -24,16 +35,33 @@ def __init__(self, data_path=os.path.join(os.path.dirname(__file__),
             # GUnzip, Parse and save MNIST data as .npy
             MnistDataset.decompress_convert_save(
                 download_path=os.path.join(os.path.dirname(__file__), 'temp'),
-                save_path=os.path.dirname(data_path))
+                save_path=data_path)
         self.data = np.load(data_path, allow_pickle=True)
 
-    # ToDo: Populate this method with a proper wget download from MNIST website
     @staticmethod
     def download_mnist(path=os.path.join(os.path.dirname(__file__), 'temp')):
-        pass
+        import urllib.request
+        import urllib.error
+
+        for file in MnistDataset.files:
+            err = None
+            for mirror in MnistDataset.mirrors:
+                try:
+                    # Disabling security linter and using hardcoded
+                    # URLs specified above
+                    url = f"{mirror}{file}"
+                    assert url.lower().startswith("http")
```

Comment:
After python JITs the code asserts are removed. You will get a bandit security lint issue that references this.

Instead use an if/else with else throwing an `AssertError` that describes the fact that `url` does not start with 'http'